app/src/main/java/com/kms/cura/view/fragment/RegDoc_AccountFragment.java

```diff
@@ -0,0 +1,176 @@
+
+    public RegDoc_AccountFragment() {
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View myFragmentView = inflater.inflate(R.layout.fragment_reg_doc__account, container, false);
+        edtPhone = setupEditText(myFragmentView, R.id.edtPhone, this);
+        edtEmail = setupEditText(myFragmentView, R.id.edtEmail, this);
+        edtPwd = setupEditText(myFragmentView, R.id.edtPwd, this);
+        edtRePwd = setupEditText(myFragmentView, R.id.edtRePwd, this);
+        btnRegister = (Button) myFragmentView.findViewById(R.id.btnRegister);
+        btnRegister.setOnClickListener(this);
+        btnBack = (ImageButton) myFragmentView.findViewById(R.id.btnRegisterBack);
+        btnBack.setOnClickListener(this);
+        edittedPwd = false;
+        edittedRePwd = false;
+        reformData();
+        return myFragmentView;
+    }
+
+    private EditText setupEditText(View parent, int resourceId, TextWatcher textWatcher) {
+        EditText edt = (EditText) parent.findViewById(resourceId);
+        edt.addTextChangedListener(textWatcher);
+        return edt;
+    }
+
+    @Override
+    public void onClick(View v) {
+        if (v.getId() == R.id.btnRegisterBack) {
+            Bundle bundle = createBundle();
+            Fragment professional = new RegDoc_ProfessionalFragment();
+            professional.setArguments(bundle);
+            getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.Fragment_DocReg, professional).commit();
+        } else if (v.getId() == R.id.btnRegister) {
+            if (checkInput()) {
+                String phone = edtPhone.getText().toString();
+                String email = edtEmail.getText().toString();
+                String pwd = edtPwd.getText().toString();
+                Bundle bundle = getArguments();
+                // check in database
+
+            }
+        }
+    }
+
+    private boolean checkInput() {
+        return checkEmail() && checkPwd() && checkRePwd() && checkPhone();
+    }
+
+    private boolean checkPhone() {
+        return !edtPhone.getText().toString().isEmpty();
+    }
+
+    private boolean checkEmail() {
+        String email = edtEmail.getText().toString();
+        return InputUtils.isEmailValid(email);
+    }
+
+    private boolean checkPwd() {
+        String pwd = edtPwd.getText().toString();
+        return InputUtils.isPasswordValid(pwd);
+    }
+
+    private boolean checkRePwd() {
+        String pwd = edtPwd.getText().toString();
+        String rePwd = edtRePwd.getText().toString();
+        return pwd.equals(rePwd);
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
```

Comment:
This method is too long and have a lot of repetitive code. Should try to reduce and re-use code here